### PR TITLE
Adapting to new package structure.

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

**Comment by anuchandy** [comment]
Few things I wanted to highlight: 

1. The main reason for the name `client-runtime` was to communicate that it has nothing specific to azure. Now we’ve the name `azure-common` so non-azure SDKs builders will be bit reluctant or confused when using it. 
2. Moved `AzureEnvironment` to `azure-common`. Built-in Environment types exposes endpoints for both Azure ARM and Data plane services.
3. The `azure-common-auth` module used to take dependency on `azure-common-mgmt` now it depends on `azure-common` (This move required doing #2)

Please share thoughts on this.

**Comment by JonathanGiles** [comment]
For easier review, people may consider browsing Anu's branch here that contains the new structure: https://github.com/anuchandy/autorest-clientruntime-for-java/tree/package-restruct

File: azure-common-auth/pom.xml [hunk 1]
```diff
@@ -48,8 +49,8 @@
 
   <dependencies>
     <dependency>
-      <groupId>com.microsoft.azure.v3</groupId>
-      <artifactId>azure-client-runtime</artifactId>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-common</artifactId>
       <version>3.0.0-SNAPSHOT</version>
```
Comment (JonathanGiles) [inline azure-common-auth/pom.xml]: We should consider rolling back the versions to 1.0.0 prior to release. Thoughts?

File: pom.xml [hunk 2]
```diff
@@ -6,13 +6,13 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
-  <groupId>com.microsoft.azure.v3</groupId>
-  <artifactId>autorest-clientruntime-for-java</artifactId>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-common-for-java</artifactId>
```
Comment (JonathanGiles) [inline pom.xml]: I would propose we call this `azure-common-parent`

Comment (anuchandy) [reply pom.xml]: will call it `azure-common-parent`

File: azure-common-auth/pom.xml (hunk 1 above)
Comment (anuchandy) [reply azure-common-auth/pom.xml]: sure, 3.x.x is not valid any more. May be we can start with `0.1.0` to indicate that it's in preview?

File: pom.xml (hunk 2 above)
Comment (JonathanGiles) [reply pom.xml]: Let's just go with 1.0.0-SNAPSHOT. If we then want to do preview, we will [version it according to the Java spec](https://azuresdkspecs.z5.web.core.windows.net/JavaSpec.html#sec-preview-releases) (so, 1.0.0-alpha1, 1.0.0-beta1, or 1.0.0-rc1).

Comment (anuchandy) [reply pom.xml]: just pushed with 0.1.0, let me change it to 1.0.0-SNAPSHOT. Thanks for the reference.

File: azure-common-auth/pom.xml (hunk 1 above)
Comment (anuchandy) [reply azure-common-auth/pom.xml]: Going with 1.0.0-SNAPSHOT as discussed below.

**Pull request merged**